rust/src/block/previous_state_hash.rs

```diff
@@ -23,31 +22,25 @@ pub struct PreviousStateHash(pub String);
 impl PreviousStateHash {
     pub fn from_path(path: &Path) -> anyhow::Result<Self> {
         let file = File::open(path)?;
-        let mut reader = BufReader::new(file);
-
-        // Read the entire file into a buffer (assuming it's large, but you only need
-        // the start)
-        let mut buffer = String::new();
-        reader.read_to_string(&mut buffer)?;
-
-        // Locate "protocol_state" within the JSON
-        if let Some(start_pos) = buffer.find("\"protocol_state\":") {
-            if let Some(brace_pos) = buffer[start_pos..].find('{') {
-                let actual_start = start_pos + brace_pos;
-
-                // Extract a fixed length portion of the buffer after the "protocol_state"
-                let slice = &buffer[actual_start..actual_start + 77];
-                let slice_with_brace = format!("{}{}", slice, "}"); // Add the closing brace manually
-
-                // Deserialize just this portion
-                let v: Value = serde_json::from_str(&slice_with_brace)?;
-
-                // Extract "previous_state_hash"
-                if let Some(previous_state_hash) = v.get("previous_state_hash") {
-                    // Convert to string and return
-                    if let Some(hash_str) = previous_state_hash.as_str() {
-                        return Ok(Self(hash_str.to_string()));
-                    }
+        let reader = BufReader::new(file);
+
+        // Create a buffer with a capacity of 1Kb
+        let mut buffer = String::with_capacity(1000);
+
+        // Limit the reader to read only the first 300 bytes
+        reader.take(1000).read_to_string(&mut buffer)?;
+
+        // Locate "previous_state_hash" within the buffer
+        let prev_state_hash_key = "\"previous_state_hash\"";
+        if let Some(hash_pos) = buffer.find(prev_state_hash_key) {
+            let hash_start = hash_pos + prev_state_hash_key.len();
+            // Find the first quote after the colon
+            if let Some(quote_start) = buffer[hash_start..].find('"') {
+                let actual_start = hash_start + quote_start + 1; // Move to the start of the hash
+                                                                 // Find the ending quote
+                if let Some(quote_end) = buffer[actual_start..].find('"') {
```

Comment:
The hash is fixed size so you shouldn't need to find the closing double quote

Reply:
I believe it's 52 characters wide

Reply:
Will try to address this in a follow-up PR.

Reply:
Fixed in this commit: https://github.com/Granola-Team/mina-indexer/commit/931f86f52bb55891aba064afbf302bdefc8d8d2f